File: july_22_tabular/R/baseline.R
library(tidyverse)

# Setup -------------------
df <- read_csv(here::here("july_22_tabular/data/data.csv"))

ids <- df$id

mtrx <- as.matrix(df[, 2:ncol(df)])

# K-Means ----------------

#setting function parameters
n_start <- 20
ks <- 2:10

res <- map(ks, ~kmeans(mtrx, centers = .x, nstart = n_start))

err_tbl <- tibble(
    k = ks,
    err = map_dbl(res, ~ pluck(., 5))
)

# take a look at scree plot
ggplot(err_tbl, aes(x = ks, y = err)) +
    geom_point() +
    geom_line() +
    theme_minimal()
#ok well this isn't definitive but let's go with 6

# Getting Preds -------------

sub <- tibble(
    Id = ids,
    Predicted = res[[9]]$cluster - 1
)

write_csv(sub, here::here("july_22_tabular/submissions/baseline.csv"))

File: july_22_tabular/jl/baseline.jl
using Revise
using CSV
using DataFrames
using Clustering

#read in data
df = CSV.read("july_22_tabular/data/data.csv", DataFrame)

#get rid of id
X = df[:, 2:ncol(df)]

#kmeans apparently expects a transposed matrix, like:
# Matrix(X)'

#estimate kmeans using 7 clusters
C = Clustering.kmeans(Matrix(X)', 7)

#write submission
sub = DataFrame(
    Id = df[!, :id],
    Predicted = C.assignments
)

CSV.write("july_22_tabular/submissions/baseline_jl.csv", sub)